crates/services/txpool/src/error.rs

```diff
@@ -0,0 +1,166 @@
+    NotInsertedInputContractDoesNotExist(ContractId),
+    #[display(fmt = "Transaction is not inserted. ContractId is already taken {_0:#x}")]
+    NotInsertedContractIdAlreadyTaken(ContractId),
+    #[display(fmt = "Transaction is not inserted. BlobId is already taken {_0:#x}")]
+    NotInsertedBlobIdAlreadyTaken(BlobId),
+    #[display(fmt = "Transaction is not inserted. UTXO does not exist: {_0:#x}")]
+    NotInsertedInputUtxoIdNotDoesNotExist(UtxoId),
+    #[display(fmt = "Transaction is not inserted. UTXO is spent: {_0:#x}")]
+    NotInsertedInputUtxoIdSpent(UtxoId),
+    #[display(
+        fmt = "Transaction is not inserted. Message id {_0:#x} does not match any received message from the DA layer."
+    )]
+    NotInsertedInputMessageUnknown(Nonce),
+    #[display(
+        fmt = "Transaction is not inserted. UTXO requires Contract input {_0:#x} that is priced lower"
+    )]
+    NotInsertedContractPricedLower(ContractId),
+    #[display(
+        fmt = "Transaction is not inserted. Input coin mismatch the values from database"
+    )]
+    NotInsertedIoCoinMismatch,
+    #[display(
+        fmt = "Transaction is not inserted. Input output mismatch. Coin owner is different from expected input"
+    )]
+    NotInsertedIoWrongOwner,
+    #[display(
+        fmt = "Transaction is not inserted. Input output mismatch. Coin output does not match expected input"
+    )]
+    NotInsertedIoWrongAmount,
+    #[display(
+        fmt = "Transaction is not inserted. Input output mismatch. Coin output asset_id does not match expected inputs"
+    )]
+    NotInsertedIoWrongAssetId,
+    #[display(
+        fmt = "Transaction is not inserted. Input message mismatch the values from database"
+    )]
+    NotInsertedIoMessageMismatch,
+    #[display(
+        fmt = "Transaction is not inserted. Input output mismatch. Expected coin but output is contract"
+    )]
+    NotInsertedIoContractOutput,
+    #[display(
+        fmt = "Transaction is not inserted. Maximum depth of dependent transaction chain reached"
+    )]
+    NotInsertedMaxDepth,
+    // small todo for now it can pass but in future we should include better messages
+    #[display(fmt = "Transaction removed.")]
+    Removed,
+    #[display(
+        fmt = "Transaction expired because it exceeded the configured time to live `tx-pool-ttl`."
+    )]
+    TTLReason,
+    #[display(fmt = "Transaction squeezed out because {_0}")]
+    SqueezedOut(String),
+    #[display(fmt = "Invalid transaction data: {_0:?}")]
+    ConsensusValidity(CheckError),
+    #[display(fmt = "Mint transactions are disallowed from the txpool")]
+    MintIsDisallowed,
+    #[display(fmt = "The UTXO `{_0}` is blacklisted")]
+    BlacklistedUTXO(UtxoId),
+    #[display(fmt = "The owner `{_0}` is blacklisted")]
+    BlacklistedOwner(Address),
+    #[display(fmt = "The contract `{_0}` is blacklisted")]
+    BlacklistedContract(ContractId),
+    #[display(fmt = "The message `{_0}` is blacklisted")]
+    BlacklistedMessage(Nonce),
```

Comment:
not a review comment, but rather a question
is this for regulatory purposes? @xgreenx

Reply:
No, it is for the case of bugs in the network that causes money printing=)